### PR TITLE
rust-cache: add cargo cache in rust actions #631

### PR DESCRIPTION
**Pull request by AsadullahFarooqi**

### CODE REVIEW

**Comment by AsadullahFarooqi** [comment]
@eddyxu

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -15,6 +15,17 @@ jobs:
       run:
         working-directory: ./rust
     steps:
+      - uses: actions/cache@v3
```
Comment (changhiskhan) [inline .github/workflows/rust.yml:18]: i think you need to change the trigger path so this change will trigger the rust CI jobs?

Comment (AsadullahFarooqi) [reply .github/workflows/rust.yml:18]: hmm, after doing some research it seems the paths are fine as they are also added the same in the mac-build section, @eddyxu is there any way where I can test it?

**Comment by changhiskhan** [comment]
closing due to inactivity

**Pull request closed without merging**